exonum/src/runtime/mod.rs

```diff
@@ -255,9 +258,20 @@ pub trait Runtime: Send + Debug + 'static {
         tx_sender: &ApiSender,
     );
 
-    fn services_api(&self) -> Vec<(String, ServiceApiBuilder)> {
+    /// Collect the full list of API handlers from the runtime for the built-in Exonum API server.
+    ///
+    /// This method is called during an API server restart. Use this method if you do not plan to
+    /// use your own API processing mechanism.
+    #[doc(hidden)]
+    fn api_endpoints(&self, _context: &ApiContext) -> Vec<(String, ServiceApiBuilder)> {
         Vec::new()
     }
+
+    /// Notify runtime about changes in list of available in API service instances.
+    ///
+    /// The purpose of this method is provide building blocks to create your own API processing
+    /// mechanisms.
+    fn notify_api_changes(&self, _context: &ApiContext, _changes: &[ApiChange]) {}
```

Comment:
1. What is the relative order of `api_endpoints` and `notify_api_changes`? 
Can the runtime implement both?
2. (will be more important when we get proper stop_service) What is the relative order between `stop_service` and a request to make service "A" API inaccessible?
3. (will be more important when we get proper stop_service)  Can this method be blocking?

Also, I wonder if it would make sense to combine `api_endpoints` and `notify_api_changes` into a single method that allows to either create API handlers for Exonum API server, or manage runtime-specific API connections 🤔

Reply:
`api_endpoints` is a transition method, not to break the existing `RustRuntime` code, in the future it will be replaced entirely by the `notify_api_change`.